### ISSUE TITLE
PyProj package and GDAL bindings are unavailable in Blender 4.2.2 (Ubuntu)

### ISSUE THREAD

**Issue by swiss-knight** [domlysz/BlenderGIS#928]
# **Blender and OS versions**
- Blender 4.2.2 (installed with snap)
- Ubuntu 22.04.5 LTS

# **Describe the bug**

When starting Blender, there are some missing dependencies:
```
2024-09-26:11:27:55,752 INFO    [__init__.py:92] ###### Starting new Blender session : 2024-09-26 11:27:55
2024-09-26:11:27:55,754 DEBUG   [checkdeps.py:9] GDAL Python binding unavailable
2024-09-26:11:27:55,755 DEBUG   [checkdeps.py:20] PyProj unavailable
2024-09-26:11:27:55,755 DEBUG   [checkdeps.py:34] Pillow available
2024-09-26:11:27:55,766 DEBUG   [checkdeps.py:46] ImageIO Freeimage plugin available
```

To check that, I also tried to import `pyproj` from the Blender scripting environment:

```py
>>> import pyproj
Traceback (most recent call last):
  File "/snap/blender/5374/4.2/python/lib/python3.11/code.py", line 90, in runcode
    exec(code, self.locals)
  File "<blender_console>", line 1, in <module>
ModuleNotFoundError: No module named 'pyproj'
```

# **How to Reproduce**
Simply start Blender after having installed BlenderGIS

# **Error message**

See above.

**Comment by swiss-knight** [comment]
Okay, I finally figure it out.

One has to install the missing packages manually because they are not embed into BlenderGIS:

```sh
#!/bin/bash

# Adapt this variable to your needs; this has to match the Blender installation folder:
BLENDER_HOME="/home/username/blender/"

# Move into the python/bin subdirectory:
cd ${BLENDER_HOME}/4.2/python/bin

# Upgrade setuptools and wheel packages:
$ ./python3.11 -m pip install -U setuptools
$ ./python3.11 -m pip install -U wheel

# Install the packages you want unless they need specific C++ lib:
$ ./python3.11 -m pip install pyproj

# For GDAL which needs specific C++ lib to be compiled:
## First: figure out GDAL version using 
$ gdalinfo --version
GDAL 3.4.1, released 2021/12/27 # <-- here: 3.4.1

## Then, this is really important, install the Python headers of the exact same python version 
## than the one embedded in Blender:
$ apt install libpython3.11-dev

## After that, symlink them to the Python folder of Blender installation dir:
$ mkdir -p ${BLENDER_HOME}/4.2/python/include
$ ln -s  /usr/include/python3.11 ${BLENDER_HOME}/4.2/python/include/python3.11/

## Finally install GDAL with the same version flag as the one given by gdalinfo --version before
$ ./python3.11 -m pip install gdal==3.4.1
Collecting gdal==3.4.1
  Using cached GDAL-3.4.1.tar.gz (755 kB)
  Preparing metadata (setup.py) ... done
Building wheels for collected packages: gdal
  Building wheel for gdal (setup.py) ... done
  Created wheel for gdal: filename=GDAL-3.4.1-cp311-cp311-linux_x86_64.whl size=4236286 sha256=6178332f1b6a97a3cb4a6e74ad45685ce21ad1ea0616dd5920c2b0816440e43e
  Stored in directory: /home/<username>/.cache/pip/wheels/e9/71/fd/44c1a8ffcf965090e76f303dee7ee88bf5c3ec34d5d2803c90
Successfully built gdal
Installing collected packages: gdal
Successfully installed gdal-3.4.1

```

**Comment by master0v** [comment]
on a mac I also had to do the following to allow me to modify the built-in blender python:

`sudo xattr -cr ~/Downloads/Blender.app`

then run the following command to check that it removed all the quarantine attributes

`xattr ~/Downloads/Blender.app`

If it didn't work, delete, and re-download Blender

also 

```
export CPLUS_INCLUDE_PATH=/opt/homebrew/include
export C_INCLUDE_PATH=/opt/homebrew/include
export LIBRARY_PATH=/opt/homebrew/lib
./python3.11 -m pip install gdal
```
